### PR TITLE
Downgrade the version of Apache Curator from 5.5.0 to 5.3.0 to avoid a bug in the new version

### LINKED ISSUE #16411: 2 Coordinators Elected Leader
Please provide a detailed title (e.g. "Broker crashes when using TopN query with Bound filter" instead of just "Broker crashes").

### Affected Version

28.0.1 (also observed in v25)
ZK version 3.7

### Description

During patching of our underlying EKS nodes, we observe a condition wherein 2 coordinators are elected leader. When we encounter this condition, we see multiple task failures across different data sources.

image

### PR DESCRIPTION
**Pull request by asdf2014**
Fixes #16411

#### Release note

<hr>

This PR has:

- [x] been self-reviewed.
   - [x] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [x] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [x] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [x] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [x] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by cryptoe** [comment]
I think this is important enough to hold the release for druid 30.

**Review by cryptoe — APPROVED** [review]

**Comment by gianm** [comment]
Are there any fixes in Curator 5.4 or 5.5 for other bugs that would be important for us? I ask since if yes- we could stay with the newer Curator and work around this bug, such as by closing and recreating our LeaderLatch when ZK session changes.

File: pom.xml [hunk 1]
```diff
@@ -75,7 +75,7 @@
         <java.version>8</java.version>
         <project.build.resourceEncoding>UTF-8</project.build.resourceEncoding>
         <aether.version>0.9.0.M2</aether.version>
-        <apache.curator.version>5.5.0</apache.curator.version>
+        <apache.curator.version>5.3.0</apache.curator.version>
```
Comment (gianm) [inline pom.xml:78]: There could be a comment here about why we're in this version, so the rationale won't get forgotten. (Link to the Curator JIRA is best)

**Comment by gianm** [comment]
I haven't reviewed these lists yet but here they are-

- 5.4 https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314425&version=12352051
- 5.5 https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314425&version=12352495
- 5.6 https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12314425&version=12353185

**Comment by gianm** [comment]
Just reviewed the lists. By downgrading from 5.5 to 5.3 we do lose various fixes. These ones sound like they could be important:

- [[CURATOR-504](https://issues.apache.org/jira/browse/CURATOR-504)] - Race conditions in LeaderLatch after reconnecting to ensemble
- [[CURATOR-638](https://issues.apache.org/jira/browse/CURATOR-638)] - Curator disconnect from zookeeper when IPs change [seems especially relevant to k8s environments]
- [[CURATOR-644](https://issues.apache.org/jira/browse/CURATOR-644)] - CLONE - Race conditions in LeaderLatch after reconnecting to ensemble [a live-lock issue; we believe the fix for this bug introduced the split-brain problem; so rolling back would reintroduce the live-lock issue]
- [[CURATOR-649](https://issues.apache.org/jira/browse/CURATOR-649)] - Background exception was not retry-able or retry gave up [robustness]

With regard to downgrading Curator to 5.3 in Druid 30, I think we should be especially careful of these issues, and in particular CURATOR-638 and possible impact on k8s environments.

Alternate approaches do include:

- Stay with Curator 5.5 and work around this bug, such as by closing and recreating our LeaderLatch when ZK session changes.
- Stay with Curator 5.5 and _don't_ work around this bug; wait for release of Curator 5.7 which will hopefully include a fix.

File: pom.xml (hunk 1 above)
Comment (asdf2014) [reply pom.xml:78]: @gianm Done. FYI, https://github.com/apache/druid/pull/16444

**Pull request merged**